client/my-sites/email/email-management/home/email-mailbox-warnings.jsx

```diff
@@ -0,0 +1,167 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { Button } from '@automattic/components';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'calypso/components/gridicon';
+import { resendVerificationEmail } from 'calypso/state/email-forwarding/actions';
+import {
+	getEmailForwardAddress,
+	hasGoogleAccountTOSWarning,
+	isEmailUserAdmin,
+	isGoogleEmailAccount,
+} from 'calypso/lib/emails';
+import { getGoogleAdminUrl } from 'calypso/lib/gsuite';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import {
+	EMAIL_WARNING_SLUG_GOOGLE_MAILBOX_TOS,
+	EMAIL_WARNING_SLUG_UNVERIFIED_FORWARD,
+} from 'calypso/lib/emails/email-provider-constants';
+import { isEmailForwardAccount } from 'calypso/lib/emails/is-email-forward-account';
+
+const MailboxListItemWarning = ( { warningText } ) => {
+	return (
+		<div className="email-mailbox-warnings__warning">
+			<Gridicon icon="info-outline" size={ 18 } />
+			<span>{ warningText }</span>
+		</div>
+	);
+};
+
+const MailboxListItemAction = ( { buttonText, external, ...otherProps } ) => {
+	return (
+		<div className="email-mailbox-warnings__action">
+			<Button compact { ...otherProps }>
+				<span>{ buttonText }</span>
+				{ external && <Gridicon icon="external" /> }
+			</Button>
+		</div>
+	);
+};
+
+const resendEmailForwardVerification = ( mailbox, dispatch ) => {
+	const destination = getEmailForwardAddress( mailbox );
+	dispatch(
+		recordTracksEvent(
+			'calypso_email_management_email_forwarding_resend_verification_email_click',
+			{
+				destination,
+				domain_name: mailbox.domain,
+				mailbox: mailbox.mailbox,
+			}
+		)
+	);
+	dispatch( resendVerificationEmail( mailbox.domain, mailbox.mailbox, destination ) );
+};
+
+const getDetailsForWarning = ( { account, dispatch, mailbox, translate, warning } ) => {
+	const warningSlug = warning.warning_slug;
+
+	if ( isGoogleEmailAccount( account ) ) {
+		if ( warningSlug === EMAIL_WARNING_SLUG_GOOGLE_MAILBOX_TOS ) {
+			const finishSetupForGoogle = {
+				actionProps: {
+					buttonText: translate( 'Finish setup' ),
+					external: true,
+					href: getGoogleAdminUrl( mailbox.domain ),
+					target: '_blank',
+				},
+				warningText: translate( 'Action required' ),
+			};
+
+			if ( isEmailUserAdmin( mailbox ) ) {
+				return finishSetupForGoogle;
+			}
+
+			// For non-admin users we disable the button if the account has been suspended
+			// due to not accepting the account ToS.
+			const googleAccountSuspendedDueToTOS = hasGoogleAccountTOSWarning( account );
+
+			return {
+				actionProps: {
+					...finishSetupForGoogle.actionProps,
+					disabled: googleAccountSuspendedDueToTOS,
+					href: googleAccountSuspendedDueToTOS ? null : finishSetupForGoogle.actionProps.href,
```

Comment:
We actually show the `Finish Setup` button for non-admin users, we should invert all those conditions.

Reply:
😳 I definitely got this one wrong!
Your proposed logic for the button in [this comment](https://github.com/Automattic/wp-calypso/pull/53032#pullrequestreview-664396027) avoids this issue. so I've gone down that path.